nautobot_golden_config/models.py

```diff
@@ -281,16 +281,23 @@ class ConfigCompliance(PrimaryModel):
     intended = models.JSONField(blank=True, help_text="Intended Configuration for feature")
     missing = models.JSONField(blank=True, help_text="Configuration that should be on the device.")
     extra = models.JSONField(blank=True, help_text="Configuration that should not be on the device.")
+    pushed = models.JSONField(blank=True, help_text="Configuration that should be pushed to the device.")
     ordered = models.BooleanField(default=True)
     # Used for django-pivot, both compliance and compliance_int should be set.
     compliance_int = models.IntegerField(null=True, blank=True)
 
     csv_headers = ["Device Name", "Feature", "Compliance"]
+    insert_secrets = False
 
     def get_absolute_url(self):
         """Return absolute URL for instance."""
         return reverse("plugins:nautobot_golden_config:configcompliance", args=[self.pk])
 
+    def get_pushed(self):
```

Comment:
`pushed`: I only see this necessary when there is "remediation", in full config we could take the "intended" as a good reference. To take this into account, I have changed the diagram (if there is no remediation, `pushed` == `intended`):

![image](https://user-images.githubusercontent.com/15998111/181174157-8974cde4-c1be-4edc-812c-e438008eb125.png)


The logic to insert secrets I would like to expose to the operator via Jinja syntax, using the very same filter that it's already created for secrets. The trick is that instead of be rendered in the first "intended" Nornir task, it would be simply ignored, and rendered as a variable for the second one. Similar to what we do in [cookiecutter](https://github.com/networktocode-llc/cookiecutter-ntc/blob/main/nautobot-plugin/%7B%7B%20cookiecutter.project_slug%20%7D%7D/.github/workflows/ci.yml#L80). So the magic itself it's basically, take the `pushed` config, either is the `intended` one or if it has gone through a remediation calculation, and inject the secrets using jinja rendering.

About the question, "how" to trigger the instert secrets, we need to pass the `request.user` context to the Jinja rendering, to properly match the user permissions to the secrets objects. It's not black or white, so maybe a user has permissions to render the device secrets, but not the circuits ones, for example.

Maybe it could be as simple as the `get_pushed()` takes as input argument the `user` that is requesting it, something like:

```py
def get_pushed_with_secrets(self, user):
    return secrets_rendering(self.pushed, user)
```

and the `secrets_rendering` it's what it's on the proposed PR to rendering with jinja, but we will find a configuration rendered at 99%, and only with few secrets filter to actually render, taking into account the `user` that it's requesting it.

My point is to call this `get_pushed_with_secrets` from a APIView for that purpose, and eventually from the `config_push` job.